packages/perseus/src/widgets/radio/radio-component.new.tsx

```diff
@@ -56,85 +52,58 @@ export type RadioChoiceWithMetadata = PerseusRadioChoice & {
-class Radio extends React.Component<Props> implements Widget {
-    static contextType = PerseusI18nContext;
-    declare context: React.ContextType<typeof PerseusI18nContext>;
-
-    // @ts-expect-error - TS2564 - Property 'focusFunction' has no initializer and is not definitely assigned in the constructor.
-    focusFunction: FocusFunction;
-
-    static defaultProps: DefaultProps = {
-        choices: [],
-        multipleSelect: false,
-        countChoices: false,
-        deselectEnabled: false,
-        linterContext: linterContextDefault,
-        showSolutions: "none",
-    };
-
-    static getUserInputFromProps(
-        props: Props,
-        unshuffle: boolean = true,
-    ): PerseusRadioUserInput {
-        // Return checked inputs in the form {choicesSelected: [bool]}. (Dear
-        // future timeline implementers: this used to be {value: i} before
-        // multiple select was added)
-        if (props.choiceStates) {
-            const choiceStates = props.choiceStates;
-            const choicesSelected = choiceStates.map(() => false);
-
-            for (let i = 0; i < choicesSelected.length; i++) {
-                const index = unshuffle ? props.choices[i].originalIndex : i;
-
-                choicesSelected[index] = choiceStates[i].selected;
-            }
-
-            return {
-                choicesSelected,
-            };
-            // Support legacy choiceState implementation
-        }
-        /* c8 ignore if - props.values is deprecated */
-        const {values} = props;
-        if (values) {
-            const choicesSelected = [...values];
-            const valuesLength = values.length;
-
-            for (let i = 0; i < valuesLength; i++) {
-                const index = unshuffle ? props.choices[i].originalIndex : i;
-                choicesSelected[index] = values[i];
-            }
-            return {
-                choicesSelected,
-            };
-        }
-        // Nothing checked
-        return {
-            choicesSelected: props.choices.map(() => false),
-        };
-    }
-
-    componentDidUpdate(prevProps: Props) {
-        if (
-            this.props.showSolutions === "selected" &&
-            prevProps.showSolutions !== "selected"
-        ) {
-            this.showRationalesForCurrentlySelectedChoices(this.props);
-        }
-    }
+const RadioComponent = (props: Props) => {
+    const {
+        choices,
+        multipleSelect = defaultProps.multipleSelect,
+        countChoices = defaultProps.countChoices,
+        deselectEnabled = defaultProps.deselectEnabled,
+        linterContext = defaultProps.linterContext,
+        showSolutions = defaultProps.showSolutions,
+        static: isStatic,
+        reviewModeRubric,
+        reviewMode,
+        apiOptions,
+        isLastUsedWidget,
+        questionCompleted,
+        findWidgets,
+        onChange,
+        trackInteraction,
+        numCorrect,
+    } = props;
+
+    const context = useContext(PerseusI18nContext);
```

Comment:
I didn't see any other usage of context in this file, we can simplify it by:
```suggestion
    const {strings} = useContext(PerseusI18nContext);
```